packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.tsx

```diff
@@ -121,6 +120,12 @@ const TabularDetailPage = (props: TTabularDetailPageProps) => {
           }
         />
       </TabularPageContainer>
+      <CustomViewsSelectorWrapper>
+        <CustomViewsSelector
+          margin={`${uiKitDesignTokens.spacing30} 0 0 0`}
```

Comment:
Is there a different approach to avoid passing margin explicitly? Shouldn't margins / spacings be determined by the parent container?

I might be missing something... 🤔

Reply:
That's what I first tried but I couldn't make it work. The main problem is that the `CustomViewsSelector` does not not if the component will be rendered at all as it depends on the request it will send to the backend in order to get the configured Custom Views for the provided locator. If there are not Custom Views configured, the `CustomViewsSelector` component will not be rendered but the consumer margins would be applied so an extra white space would be rendered.

Reply:
If the `CustomViewsSelector` returns `null`, there shouldn't be any extra margin right?

Let's maybe clarify this in our sync

Reply:
The problem is not about a the use case for a  `CustomViewsSelector` which does not receive a locator but the use case when it does receive one but does not have any Custom Views configured: since in case we do receive anything we need to render a `div` with zero height to be able to run the animation, the extra space coming from the `Spacing.Stack` is then applied.